lib/extensions/ar_migration.rb

```diff
@@ -0,0 +1,53 @@
+module MigrationSyncHelper
+  def self.migrations_column_present?
+    ActiveRecord::Base.connection.columns("miq_regions").detect { |c| c.name == "migrations_ran" }
+  end
+
+  class HelperARClass < ActiveRecord::Base; end
+  def self.restart_subscription(s)
+    c = HelperARClass.establish_connection.connection
+    c.pglogical.subscription_disable(s.id)
+    c.pglogical.subscription_enable(s.id)
+  ensure
+    HelperARClass.remove_connection
+  end
+
+  def self.wait_for_remote_region_migration(subscription, version, wait_time = 1)
+    return unless MigrationSyncHelper.migrations_column_present?
+    region = MiqRegion.find_by(:region => subscription.provider_region)
+    until region.migrations_ran&.include?(version)
+      restart_subscription(subscription)
+      sleep(wait_time)
+      region.reload
+    end
+  end
+
+  def self.update_local_migrations_ran(version, direction)
+    return unless MigrationSyncHelper.migrations_column_present?
+    return unless (region = MiqRegion.my_region)
+
+    new_migrations = ActiveRecord::SchemaMigration.normalized_versions
+    new_migrations << version if direction == :up
+    migrations_value = ActiveRecord::Base.connection.quote(PG::TextEncoder::Array.new.encode(new_migrations))
+
+    ActiveRecord::Base.connection.exec_query(<<~SQL)
+      UPDATE #{MiqRegion.table_name}
+      SET migrations_ran = #{migrations_value}
+      WHERE id = #{region.id}
+    SQL
+  end
+end
+
+module ArPglogicalMigration
+  def migrate(direction)
+    PglogicalSubscription.all.each do |s|
+      MigrationSyncHelper.wait_for_remote_region_migration(s, version.to_s)
+    end
+
+    ret = super
+    MigrationSyncHelper.update_local_migrations_ran(version.to_s, direction)
+    ret
```

Comment:
```ruby
super.tap do 
  MigrationSyncHelper.update_local_migrations_ran(version.to_s, direction)
end
```

might read a little better than having a temp var.

Reply:
@bdunne actually suggested the same thing and I prefer the temp variable because I see the "return the original object" part of `tap`s behavior as more of a side-effect rather than the primary reason for using it. If my goal is to just return the same value rather than do something with the value returned by `super` I like this way better.

Reply:
I agree with not using tap since we're not actually using the proposed block variable.

Reply:
Although, what does `super` return?  `ret` could be named better.  I honestly have no idea what it is.

Reply:
Oh funny, I see .tap's return of the original value as the main feature...in that sense I read it as more of a "oh and by the way", and it happens to just have a reference to the tapped thing, but it's not required to use it.

Reply:
Same here, "I have a thing that I need to return, but before I return it, I need to do this other thing"

> I agree with not using tap since we're not actually using the proposed block variable.

@jrafanie we use block syntax all the time without referencing anything that is yielded

Reply:
yeah, I can see that use case.  I prefer the original for clear separation of the super from the post-super work.  It's preference.